api/genericresource/helpers.go

```diff
@@ -0,0 +1,122 @@
+package genericresource
+
+import (
+	"fmt"
+
+	"github.com/docker/swarmkit/api"
+)
+
+func newError(format string, args ...interface{}) error {
+	return fmt.Errorf("could not parse GenericResource: "+format, args...)
+}
+
+// NewSet creates a set object
+func NewSet(key string, vals ...string) []*api.GenericResource {
+	rs := make([]*api.GenericResource, 0, len(vals))
+
+	for _, v := range vals {
+		rs = append(rs, NewString(key, v))
+	}
+
+	return rs
+}
+
+// NewString creates a String resource
+func NewString(key, val string) *api.GenericResource {
+	return &api.GenericResource{
+		Resource: &api.GenericResource_Str{
+			Str: &api.GenericString{
+				Kind:  key,
+				Value: val,
+			},
+		},
+	}
+}
+
+// NewDiscrete creates a Discrete resource
+func NewDiscrete(key string, val int64) *api.GenericResource {
+	return &api.GenericResource{
+		Resource: &api.GenericResource_Discrete{
+			Discrete: &api.GenericDiscrete{
+				Kind:  key,
+				Value: val,
+			},
+		},
+	}
+}
+
+// NewResources creates a Resources object
+func NewResources() *api.Resources {
+	return &api.Resources{
+		Generic: make([]*api.GenericResource, 0),
+	}
+}
+
+func getKind(res *api.GenericResource) string {
+	switch r := res.Resource.(type) {
+	case *api.GenericResource_Discrete:
+		return r.Discrete.Kind
+	case *api.GenericResource_Str:
+		return r.Str.Kind
+	}
+
+	return ""
+}
+
+// GetResource returns resources from the "resources" parameter matching the kind key
+func GetResource(kind string, resources []*api.GenericResource) (res []*api.GenericResource) {
```

Comment:
The named return `(res []*api.GenericResource)` is inconsistent with other functions?